apps/zk-explorer/components/SidebarFilters/SidebarFilters.tsx

```diff
@@ -0,0 +1,206 @@
+import {
+  FilterCategory,
+  FilterCategoryItem,
+  FilterConstraints,
+  FilterOptionItem,
+} from './types';
+
+export type SidebarFiltersProps = PropsOf<'div'> & {
+  onConstraintsChange: (constraints: FilterConstraints) => void;
+};
+
+export const SidebarFilters: FC<SidebarFiltersProps> = ({
+  onConstraintsChange,
+}) => {
+  const [constraints, setConstraints] = useState<FilterConstraints>({
+    [FilterCategory.ProofSystem]: [],
+    [FilterCategory.Categories]: [],
+    [FilterCategory.License]: [],
+    [FilterCategory.LanguageOrFramework]: [],
+  });
+
+  // TODO: These are only for testing purposes. Remove these once the actual data is available.
+  const debugProofSystemsOptions: FilterOptionItem[] = [
+    {
+      label: 'Circom',
+      amount: 403,
+    },
+    {
+      label: 'Plonk',
+      amount: 123,
+    },
+    {
+      label: 'Halo2',
+      amount: 234,
+    },
+    {
+      label: 'Bulletproof',
+      amount: 43,
+    },
+    {
+      label: 'Stark',
+      amount: 78,
+    },
+  ];
+
+  const debugCategoryOptions: FilterOptionItem[] = [
+    {
+      label: 'Identity Verification',
+      amount: 59,
+    },
+    {
+      label: 'Private Transaction',
+      amount: 290,
+    },
+    {
+      label: 'Voting System',
+      amount: 12,
+    },
+    {
+      label: 'Arithmetic',
+      amount: 90,
+    },
+    {
+      label: 'Cryptography',
+      amount: 183,
+    },
+  ];
+
+  const debugLicenseOptions: FilterOptionItem[] = [
+    {
+      label: 'MIT',
+      amount: 392,
+    },
+    {
+      label: 'GPLv3',
+      amount: 19,
+    },
+    {
+      label: 'Apache 2.0',
+      amount: 128,
+    },
+  ];
+
+  const debugLanguageOptions: FilterOptionItem[] = [
+    {
+      label: 'TypeScript',
+      amount: 410,
+    },
+    {
+      label: 'C++',
+      amount: 319,
+    },
+    {
+      label: 'Rust',
+      amount: 593,
+    },
+    {
+      label: 'Circom',
+      amount: 478,
+    },
+    {
+      label: 'Solidity',
+      amount: 92,
+    },
+    {
+      label: 'JavaScript',
+      amount: 228,
+    },
+  ];
+
+  const categories: FilterCategoryItem[] = [
+    {
+      category: FilterCategory.ProofSystem,
+      options: debugProofSystemsOptions,
+    },
+    {
+      category: FilterCategory.Categories,
+      options: debugCategoryOptions,
+    },
+    {
+      category: FilterCategory.License,
+      options: debugLicenseOptions,
+    },
+    {
+      category: FilterCategory.LanguageOrFramework,
+      options: debugLanguageOptions,
+    },
+  ];
+
+  useEffect(() => {
+    onConstraintsChange(constraints);
+
+    // Under no circumstances should the `onConstraintsChange` prop
+    // function be called if the constraints themselves have not changed;
+    // therefore, `onConstraintsChange` should not be included in the
+    // effect's dependency array.
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [constraints]);
+
+  const handleConstraintChange = (
```

Comment:
add useCallback here

Reply:
The function isn't directly passed down as a prop to a child component, should I still proceed to wrap it in `useCallback`?